src/foam/nanos/auth/HumanNameTrait.js

```diff
@@ -32,12 +36,33 @@ foam.INTERFACE({
       documentation: 'Full legal name of user. Appends first, middle & last name.',
       transient: true,
       expression: function(firstName, middleName, lastName) {
-        return middleName != '' ? firstName + ' ' + middleName + ' ' + lastName : firstName + ' ' + lastName;
+        if ( firstName && middleName && lastName ) {
+          return `${firstName} ${middleName} ${lastName}`;
+        }
+
+        if ( firstName && lastName ) {
+          return `${firstName} ${lastName}`;
+        }
+
+        return '';
```

Comment:
You could also just do: return [firstName, middleName, lastName].filter(n=>n).join(' ')

Reply:
@kgrgreer this doesn't guarantee which field is not an empty string though. For instance, only firstName might be available.

Reply:
What do you mean? It just concatenates the values and includes spaces between non-null values.

Reply:
@kgrgreer Yes, but shouldn't the values be concatenated under certain conditions? For example, if firstName and middleName are non-null, and lastName is null, the function will return firstName + middleName. In this case, don't we just return an empty string?

Reply:
@kgrgreer Or we don't need to worry about the example case above because the example above is a validation error?